### PR TITLE
Backport to opendistro-0.9

### PR DESCRIPTION
**Pull request by cliu123**
##  opendistro-for-elasticsearch/security pull request intake form
_Please provide as much details as possible to get feedback/acceptance on your PR quickly_

 
 1. __Category:__ (_Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation_)
 
 
 
 2. __Github Issue # or road-map entry, if available:__

 3. __Description of changes:__

 4. __Why these changes are required?__ 

 5. __What is the old behavior before changes and new behavior after changes?__ (_Please add any example/logs/screen-shot if available_)

 6. __Testing done:__ (_Please provide details of testing done: Unit testing, integration testing and manual testing_)
 
 
 
 7. __TO-DOs, if any:__ (_Please describe pending items and provide Github issues# for each of them_)

 8. __Is it backport from master branch?__ (_If yes, please add backport PR # and commits #_)
     Cherry-pick: commit d031626b73c871de3b214f412c49e0bf479896e4
                           commit d17ecb323100a48efd1e2c70ae94fee390bae154
                           commit 059b11d87c9084cce2dbc97a5f90db25b6e39cef

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by vrozov** [comment]
Please include cherry info both in commit message and PR.

**Comment by cliu123** [comment]
> Please include cherry info both in commit message and PR.

Done

**Review by vrozov — DISMISSED** [review]

**Review by dinusX — DISMISSED** [review]

File: src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/TLSUtilTests.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public void testSSLUtilInvalidContentLength() {
             buffer.writeByte(TLS_MAJOR_VERSION);
             buffer.writeByte(TLS_MINOR_VERSION);
             //setting content length as 0
-            buffer.writeByte(0);
+            buffer.writeShort(0);
```
Comment (vrozov) [inline src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/TLSUtilTests.java:75]: This is a separate issue that needs to be fixed on master and backported to all branches.

Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/TLSUtilTests.java:75]: Can we get this PR merged first given the tight timeline? I'll make the same change in all of other branches.

Comment (cliu123) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/TLSUtilTests.java:75]: The PR for master branch: https://github.com/opendistro-for-elasticsearch/security/pull/927

Comment (vrozov) [reply src/test/java/com/amazon/opendistroforelasticsearch/security/ssl/util/TLSUtilTests.java:75]: No, there is no release that this PR blocks.

**Comment by vrozov** [comment]
PR refers to a different commits, while commit is a cherry pick for another commit. Please update PR to include all necessary commits.

**Review by vrozov — APPROVED** [review]

**Comment by cliu123** [comment]
> PR refers to a different commits, while commit is a cherry pick for another commit. Please update PR to include all necessary commits.

Done

**Pull request merged**